com.unity.render-pipelines.high-definition/Documentation~/editing-shader-properties-at-runtime.md

```diff
@@ -0,0 +1,11 @@
+# Editing shader properties at runtime
+
+For High Definition Render Pipeline (HDRP) non-Shader Graph shaders, such as the [Lit](Lit-Shader.md), [Unlit](Unlit-Shader.md) shaders, changing a property at runtime, in some cases, does not have any effect. This is because if the property is not enabled before runtime, the specific shader variant associated with this property is not included in the shader (and is also not included when bulding a player). To include the shader variant, the proper keywords need to be enabled on the material before editing. 
```

Comment:
There is various incorrect statement here. Probe exist also for shader graph.

Here is a proposal:

Each shader feature (Keyword) use in a shader (link to the doc) produce a different shader code to compile. Each of this shader code is call a shader variant. Unity try to reduce the number of shader variant to compile by only considering the sets of kewords use in a Material. Other shader variant don't exist. When building a Player, it is the union of the all shader variant based on keyword used that is compiled.

If you try to create a Material at runtime and change proerties which affect which keywords is enabled. In case this combination was not existing when building the player, this shader variant will not exist and the rendering will fail.
This can happen if you use for example the HDRP/Lit shader and try to setup a iridecence material but don't have any ididescence properties enabled in an already existing Material.


^ So this is not good but this is what I think we should express. We should be clear that 1) This is for Unity in general, not only HDRP. Builtin have same issue. And that this is also for shader graph if you add keyword in shader graph.
Then the goal is to explain that users must enable the right keyword at runtime and that this shader variant need to exist. 

this statement can be share with URP documentation.

Reply:
Thanks, will update with a new proposal with this info !